src/OrleansCodeGenerator/GrainMethodInvokerGenerator.cs

```diff
@@ -279,10 +279,36 @@ private static MethodDeclarationSyntax GenerateInvokeMethod(Type grainType, Meth
 
             // The invoke method expects a Task<object>, so we need to upcast the returned value.
             // For methods which do not return a value, the Box extension method returns a meaningless value.
-            return new StatementSyntax[]
+            if (method.IsGenericMethod)
             {
-                SF.ReturnStatement(SF.InvocationExpression(grainMethodCall.Member((Task _) => _.Box())))
-            };
+                var sb = new System.Text.StringBuilder().Append(method.Name).Append("<");                
```

Comment:
I'd be much more comfortable if you used the Roslyn APIs to create this, something like:
```C#
SF.GenericName(
  SF.Identifier(method.Name),
  TypeArgumentList()
    .AddArguments(method.GetGenericArguments().Select(_ => SF.ParseTypeName("dynamic")).ToArray()));
```